xonsh/tools.py

```diff
@@ -941,7 +943,10 @@ def ensure_slice(x):
         else:
             raise ValueError('cannot convert {!r} to slice'.format(x))
     except TypeError:
-        raise TypeError('ensure_slice() argument must be a string or a number not {}'.format(type(x)))
+        try:
+            s = slice(*(int(i) for i in x))
+        except (TypeError, ValueError):
+            raise ValueError('cannot convert {!r} to slice'.format(x))
```

Comment:
Hmm yeah, I guess this isn't a big deal either way and is non-blocking.  It sort of depends how you look at it.  Maybe it should just retain the error type the was caught?